cxx/structureTranslationModule/translator/StructureTranslator.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+
+#pragma once
+
+#include "sc-memory/sc_memory.hpp"
+#include "sc-memory/sc_addr.hpp"
+
+namespace structureTranslationModule
+{
+using ScAddrSet = std::unordered_set<ScAddr, ScAddrHashFunc<uint32_t>>;
+
+class StructureTranslator
+{
+public:
+  explicit StructureTranslator(ScMemoryContext * context)
+    : context(context)
+  {
+  }
+
+  virtual std::string translate(ScAddr const & structAddr)
+      const = 0;
+
+  size_t hashCode() const;
+
+  bool operator<(StructureTranslator const & other) const;
+
+protected:
+  ScMemoryContext * context;
+
+  std::string getEnglishContent(ScAddr const & linkNode) const;
+
+  bool isInStructure(ScAddr const & structAddr, ScAddr const & elementAddr) const;
```

Comment:
Use just context.HelperCheckEdge insted of it

Reply:
this function can't be replaced by just context.HelperCheckEdge, several conditions must be checked here